app/models/ability.rb

```diff
@@ -0,0 +1,17 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    return unless user.present?
+
+    can :read, :all
+    can :destroy, Post, author_id: user.id
+    # comments
+    can :create, Post
+    can :create, Comment
+    can :destroy, Comment, author_id: user.id
+    return unless user.role?
```

Comment:
- [ ] Update this to `user.admin?` since the keyword admin should be able to do all of them

Reply:
@GzimAsani, from the project requirements we were supposed to add a column `role` with a value `"admin" `. Calling `user.admin?` throws an error of `undefined method admin?` as the column name is `role`. Kindly advise on that.

Reply:
Hi @alphayowakarindi, kindly leave it as it is. I thought that you have specified the admin on the user model. But since that is not the case, this way will work as well. I apologize for the confusion caused

Reply:
@GzimAsani Thank you for clarifying that. I think it would be nice for the requirement to have the column name as `admin` of `boolean` type instead of `role` so one can call `user.admin?`. That looks more readable to me.